src/strategies/hedgey-locked/index.ts

```diff
@@ -0,0 +1,48 @@
+import { BigNumber, BigNumberish } from '@ethersproject/bignumber';
+import { formatUnits } from '@ethersproject/units';
+import { Multicaller } from '../../utils';
+
+export const author = 'agentcoinorg';
+export const version = '1.0.0';
+
+const abi = [
+  'function lockedBalances(address holder, address token) view returns (uint256 lockedBalance)'
+];
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+): Promise<Record<string, number>> {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  const multi = new Multicaller(network, provider, abi, { blockTag });
+
+  for (const contract of options.contracts) {
+    for (const address of addresses) {
+      multi.call(contract + "/" + address, contract, 'lockedBalances', [address, options.token]);
+    }
+  }
```

Comment:
May face memory issues if someone pass too many contracts. could you throw an error if someone pass contracts above a limit